packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,304 @@
+import { Selection, select } from 'd3-selection'
+import { hierarchy, HierarchyNode, treemap } from 'd3-hierarchy'
+import { group, max, extent } from 'd3-array'
+import { scaleLinear, scaleThreshold } from 'd3-scale'
+import { hsl } from 'd3-color'
+import { ComponentCore } from 'core/component'
+import { SeriesDataModel } from 'data-models/series'
+import { getColor, brighter, getHexValue } from 'utils/color'
+import { getString, getNumber, isNumber } from 'utils/data'
+import { smartTransition } from 'utils/d3'
+import { TreemapConfigInterface, TreemapDefaultConfig } from './config'
+import { TreemapDatum, TreemapNode } from './types'
+import * as s from './style'
+
+const defaultNumberFormat = (value: number): string => `${value}`
```

Comment:
Maybe instead of having this as a global function, we can define it as a component method?